typing/includemod_errorprinter.ml

```diff
@@ -95,43 +95,50 @@ module Illegal_permutation = struct
     | None -> g y
     | Some _ as v -> v
 
+  type change =
+    | Transposition of int * int
+    | Primitive_coercion of string
+    | Alias_coercion of Path.t
+
   (** We extract a lone transposition from a full tree of permutations. *)
   let rec transposition_under path (coerc:Typedtree.module_coercion) =
     match coerc with
     | Tcoerce_structure(c,_) ->
-        either
-          (not_fixpoint path 0) c
-          (first_non_id path 0) c
+          (either
+             (not_fixpoint path 0) c
+             (first_non_id path 0) c
+          )
     | Tcoerce_functor(arg,res) ->
         either
           (transposition_under (InArg::path)) arg
           (transposition_under (InBody::path)) res
     | Tcoerce_none -> None
-    | Tcoerce_alias _ | Tcoerce_primitive _ ->
-        (* these coercions are not inversible, and raise an error earlier when
-           checking for module type equivalence *)
-        assert false
+    | Tcoerce_alias _ | Tcoerce_primitive _ -> None
+    (* those two coercions cannot happen at toplevel *)
```

Comment:
I don't think this comment is very useful.
I've been hit in the past by these weird coercions that are only relevant as part of a bigger structure coercion, and I think the solution should be to have two coercion types (module coercions and structure field coercions). For now, I think the code is fine but I'd rather have no comment than this unhelpful comment.